core/templates/pages/skill-editor-page/services/question-creation-service.spec.ts

```diff
@@ -46,100 +47,527 @@ describe('Question Creation Service', function() {
+          content: {
+            html: 'Question 1',
+            content_id: 'content_1'
+          },
+          interaction: {
+            answer_groups: [{
+              outcome: {
+                dest: 'outcome 1',
+                feedback: {
+                  content_id: 'content_5',
+                  html: ''
+                },
+                labelled_as_correct: true,
+                param_changes: [],
+                refresher_exploration_id: null
+              },
+              rule_specs: [{
+                inputs: {
+                  x: 10
+                },
+                rule_type: 'Equals'
+              }],
+            }],
+            confirmed_unclassified_answers: [],
+            customization_args: {},
+            default_outcome: {
+              dest: null,
+              feedback: {
+                html: 'Correct Answer',
+                content_id: 'content_2'
+              },
+              param_changes: [],
+              labelled_as_correct: false
+            },
+            hints: [
+              {
+                hint_content: {
+                  html: 'Hint 1',
+                  content_id: 'content_3'
+                }
+              }
+            ],
+            solution: {
+              correct_answer: 'This is the correct answer',
+              answer_is_exclusive: false,
+              explanation: {
+                html: 'Solution explanation',
+                content_id: 'content_4'
+              }
+            },
+            id: 'TextInput'
+          },
+          param_changes: [],
+          recorded_voiceovers: {
+            voiceovers_mapping: {
+              content_1: {}
+            }
+          },
+          written_translations: {
+            translations_mapping: {
+              content_1: {}
+            }
+          },
+          solicit_answer_details: false
+        },
+        language_code: 'en',
+        version: 1
+      };
+      var sampleQuestion = QuestionObjectFactory.createFromBackendDict(
+        sampleQuestionBackendDict);
+      spyOn(QuestionObjectFactory, 'createDefaultQuestion').and.returnValue(
+        sampleQuestion);
+      $rootScope = $injector.get('$rootScope');
+      $location = $injector.get('$location');
+    }));
+
+    it('should create question', function() {
+      var skillDiff = SkillDifficultyObjectFactory.create(
+        'skillId1', 'description', 'Easy');
+      var modalSpy = spyOn($uibModal, 'open').and.returnValue(
+        {result: Promise.resolve([skillDiff])});
+      qcs.createQuestion();
+      expect(modalSpy).toHaveBeenCalled();
     });
-    spyOn(SkillBackendApiService, 'fetchMultiSkills').and.returnValue(
-      Promise.resolve([skillBackendDict]));
-    spyOn(SkillEditorStateService, 'getGroupedSkillSummaries').and.returnValue({
-      current: [],
-      others: [skillSummaryDict]
+
+    it('should return difficulty strings', function() {
+      expect(qcs.getDifficultyString(0.3)).toEqual('Easy');
```

Comment:
Add some more cases here. This is good, but it doesn't hurt to have more checks

Reply:
Added.